internal/engine/wazevo/ssa/instructions.go

```diff
@@ -323,6 +326,9 @@ const (
 	// OpcodeIcmpImm compares an integer value with the immediate value on the given condition: `v = icmp_imm Cond, x, Y`.
 	OpcodeIcmpImm
 
+	// OpcodeIandsImm performs a bitwise AND with the immediate value and updates condition flags.
+	OpcodeIandsImm
```

Comment:
This as well. There's no concept in the flag and you are supposed to use Icmp and ExitTrueIfTrueWithCond

Reply:
See other various places for doing similar thing eg memory bound check

Reply:
Currently we can do alignment check with just one instruction for both archs

https://github.com/tetratelabs/wazero/blob/main/internal%2Fengine%2Fcompiler%2Fimpl_threads_arm64.go#L40
https://github.com/tetratelabs/wazero/blob/main/internal%2Fengine%2Fcompiler%2Fimpl_threads_amd64.go#L654

I'm not too sure how to do it with Icmp or should we use two instructions, OpcodeBmask then Icmp?

Reply:
Yes two instructions

Reply:
Yes two instructions